### ISSUE TITLE
Support for MongoDB 4.2

### ISSUE THREAD

**Issue by shollander** [flapdoodle-oss/de.flapdoodle.embed.mongo#295]
Mongo has changed the binary download options and file name format in a way the breaks the embed.mongo downloader.

1. For Windows, 2012plus in now the only option. [Feature.java](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/02d644905f0b05463be31daddcf468b1b7cb6e43/src/main/java/de/flapdoodle/embed/mongo/distribution/Feature.java) needs a new value: `ONLY_WINDOWS_2012_SERVER`, and the corresponding logic in [Paths.java](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/02d644905f0b05463be31daddcf468b1b7cb6e43/src/main/java/de/flapdoodle/embed/mongo/Paths.java) needs to be implemented.
2. The generic Linux archive has been removed. There are now various distribution-specific tar files. Perhaps the RHEL 6.2 version can be used as a generic one (`mongodb-linux-x86_64-rhel62-4.2.0.tgz`).

Those were the main issues I noticed at first glance, there might be others as well.

**Comment by Loki-Afro** [comment]
thanks for notifying, can't look into it right now but did you try if the rhel one works? a bit weird thats a tar anyway ...

**Comment by Loki-Afro** [comment]
filed an issue https://jira.mongodb.org/browse/SERVER-43036 I don't know what the difference is between all these linux tgz for now - but I can start the ubuntu mongod on my fedora system - which looks promising ...

**Comment by cycnos** [comment]
@shollander 
I had the same error on windows system. I tried to run manually the executable file in ~/.embedmongo and the error was more explicit (in my case, I was missing some DLL  from visual c++ 2015)
Maybe you'll alse have an explicit error on your CentOS ;)

@Loki-Afro 
If you need some help to develop a new version in order to supprot mongoDB v4.2  I'd be pleased to contribute (at least for the windows part)

**Comment by Loki-Afro** [comment]
@cycnos thanks for the offering, but since you installed the dll (this was broken up to 4.0 I guess) it is not practical to force the user to install an additional dll. Or do you have a more fancy solution?

Some kind of similar error would be thrown when using any tar gz since they come not bundled with any libraries here - which is really annoying when you are running your build inside a docker container or similar.

But feel free to create a PR against master.

In any way, it is rather simple to just point to the 4.2 tar gz currently ... and if this is enough in the users setup so be it. :D

**Comment by sakhtar1979** [comment]
@Loki-Afro Can you explain this part a bit more?

> In any way, it is rather simple to just point to the 4.2 tar gz currently ... and if this is enough in the users setup so be it. :D

As a user of EmbeddedMongo on Linux, Mac and Windows what exactly we need to do special for it to be compatible with Mongo 4.2? If anything. Thanks

**Comment by mgaceanu** [comment]
Any progress on this issue?

**Comment by shimizuf** [comment]
We would also be very happy about 4.2 support, it would enable us to upgrade all our DB instances.

**Comment by thecoop** [comment]
We need 4.2 support to test our usage of the $merge aggregator operator, which is new in 4.2. We can't really test this without flapdoodle 4.2 support!

**Comment by keeratsingh** [comment]
@Loki-Afro  Any updates on this issue ?

**Comment by phillipuniverse** [comment]
tl;dr - it looks like this is supported starting with version 3.1.1.

If you're coming here from Spring Boot, in Spring Boot 2.7 and the latest Flapdoodle this works. I have this in my `application-test.properties`:

```
spring.mongodb.embedded.version=4.2.13
```

In my logs I can see the version being downloaded:

<img width="1762" alt="image" src="https://user-images.githubusercontent.com/684275/169539435-6136bab4-ba69-4e14-b161-64d1a0b3406b.png">

From [the Spring Boot docs](https://docs.spring.io/spring-boot/docs/2.7.x/reference/html/data.html#data.nosql.mongodb.embedded) it references this `Version` class for supported versions to target:

https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/70273c20c76f1069811b0389a4bc161f4f5c4848/src/main/java/de/flapdoodle/embed/mongo/distribution/Version.java#L227

It looks like the first support for 4.2.13 appeared in the commit at 150ab15c4d8d10ca63086bd8b515ef3c4a6f3531 which was released in version 3.1.1.

**Comment by michaelmosmann** [comment]
It took me a while to handle this "hey let's change how to download mongodb" ..  so yes, it should work now.

**Comment by DeveloperKazan** [comment]
Hello!
I use de.flapdoodle.embed.mongo:3.4.5 with PRODUCTION MongoDB Version in Kotlin app.
But I get this output:
[RuntimeException.txt](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/files/8843787/RuntimeException.txt)


WARN  de.flapdoodle.embed.mongo.packageresolver.linux.LinuxPackageFinder because there is no package for PRODUCTION:Platform{operatingSystem=Linux, architecture=X86_64} we fall back to PRODUCTION:Platform{operatingSystem=Linux, architecture=X86_64, version=Ubuntu_20_04}

Could you give me an advice how to solve this issue?

**Comment by michaelmosmann** [comment]
@DeveloperKazan can you try to run this with this properties set: https://github.com/flapdoodle-oss/de.flapdoodle.os#run
